From 4c7b22fbba85dbed71a34ab3a4427bd078f4de7c Mon Sep 17 00:00:00 2001
From: Florian Westphal <fwestpha@redhat.com>
Date: Tue, 2 Sep 2014 10:14:26 -0400
Subject: [net] Change x86_64 add32_with_carry to allow memory operand

Message-id: <1409652920-431-4-git-send-email-fwestpha@redhat.com>
Patchwork-id: 90170
O-Subject: [PATCH 03/57] [RHEL7.1 net] net: Change x86_64 add32_with_carry to allow memory operand
Bugzilla: 1131999
RH-Acked-by: Jiri Pirko <jpirko@redhat.com>
RH-Acked-by: Vlad Yasevich <vyasevic@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>
RH-Acked-by: Jiri Benc <jbenc@redhat.com>

BZ: 1131999

commit 4405b4d635aa2c5c7eb8873696b54811531e0d08
Author: Tom Herbert <therbert@google.com>
Date:   Fri May 2 16:28:40 2014 -0700

    net: Change x86_64 add32_with_carry to allow memory operand

    Note add32_with_carry(a, b) is suboptimal, as it forces
    a and b in registers.

    b could be a memory or a register operand.

    Signed-off-by: Tom Herbert <therbert@google.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/include/asm/checksum_64.h b/arch/x86/include/asm/checksum_64.h
index 3e25a0c..7187e00 100644
--- a/arch/x86/include/asm/checksum_64.h
+++ b/arch/x86/include/asm/checksum_64.h
@@ -184,7 +184,7 @@ static inline unsigned add32_with_carry(unsigned a, unsigned b)
  asm("addl %2,%0\n\t"
      "adcl $0,%0"
      : "=r" (a)
-     : "0" (a), "r" (b));
+     : "0" (a), "rm" (b));
  return a;
 }
 
-- 
1.7.1